backend/core/ai/sql_executor.py

```
@@ -0,0 +1,320 @@
+                if normalized.startswith('--'):
+                    nl = normalized.find('\n')
+                    normalized = '' if nl == -1 else normalized[nl + 1:]
+                else:
+                    end = normalized.find('*/')
+                    normalized = '' if end == -1 else normalized[end + 2:]
+                normalized = normalized.lstrip()
+            normalized = ' '.join(normalized.split())
+            first_kw = (normalized.split(None, 1)[0] if normalized else '').upper()
+            disallowed_any = {
+                'COPY', 'DO', 'CALL', 'LISTEN', 'NOTIFY', 'EXECUTE',
+                'PG_SLEEP', 'PG_TERMINATE_BACKEND', 'COPYTO', 'COPYFROM', 'COPY',
```

Ruff (0.13.1)

Sets should not contain duplicate item `'COPY'`

Remove duplicate item

(B033)